test/useCases/getUserById.integration.test.ts

```diff
@@ -97,32 +97,36 @@ describe("getUserById", () => {
       "bichard01@example.com",
       groups.map((g) => g.name)
     )
-    const currentUserId = (await selectFromTable("users", "username", "Bichard01"))[0].id
+    const currentUser = (await selectFromTable("users", "username", "Bichard01"))[0]
     const selectedGroups = await selectFromTable("groups", undefined, undefined, "name")
     const selectedGroup = selectedGroups[0]
     const user = users.filter((u) => u.username === "Bichard02")[0]
 
-    const createUserDetails = {
+    const createUserDetails: any = {
       username: user.username,
       forenames: user.forenames,
       emailAddress: user.email,
       endorsedBy: user.endorsed_by,
       surname: user.surname,
       orgServes: user.org_serves,
-      groupId: selectedGroup.id,
       visibleForces: "001,004,",
       visibleCourts: "B01,B41ME00",
       excludedTriggers: "TRPR0001,"
     }
-
-    await createUser(connection, currentUserId, createUserDetails)
+    createUserDetails[selectedGroup.name] = "yes"
+    await createUser(connection, currentUser, createUserDetails)
     const selectedUsers = await selectFromTable("users", "email", user.email)
     const selectedUserId = selectedUsers[0].id
     const userResult = await getUserById(connection, selectedUserId)
 
     expect(isError(userResult)).toBe(false)
 
     const actualUser = userResult as Partial<User>
-    expect(actualUser.groupId).toBe(selectedGroup.id)
+
+    if (actualUser.groups) {
```

Comment:
instead, can we do expect(actualUser.groups).toNotBet(undefined) / or null?
would be quicker to debug if the test fails. otherwise in the terminal you will only see fail, which is not very informative.

Reply:
It is because the actualUser is typed as a partial user and groups is optional, therefore it will get raised as a possible null reference which is why the if there. 

I don't think 'toNotBe' exists as a matcher?

Reply:
You are right ... I think it is something like .not.toBe or something

Reply:
`expect(actualUser.groups).not.toBeUndefined()` seems to do the trick in the else, I was hoping this might remove the need for an if/else but not so.

Reply:
Review comment 5

Reply:
is it because it is actually a null maybe?